src/main/java/org/apache/commons/lang3/BooleanUtils.java

```diff
@@ -263,6 +267,23 @@ public static Boolean negate(final Boolean bool) {
         return bool.booleanValue() ? Boolean.FALSE : Boolean.TRUE;
     }
 
+    /**
+     * <p>Validate that the specified argument array is neither {@code null}
+     * nor a length of zero (no elements); otherwise throwing an exception.
+     *
+     * @param array  an array of {@code boolean}s
+     * @throws IllegalArgumentException if {@code array} is {@code null}
+     * @throws IllegalArgumentException if {@code array} is empty.
+     */
+    private static void notNullOrEmpty(final boolean... array){
+        if (array == null) {
```

Comment:
Hi @arturobernalg 
I think we should converge all null checks to `Objects.requireNonNull(arg, "arg")`, which also avoids making our exception messages too English-centric. In the same vein, I'd call this new method `requireNotEmpty`.